### DISCUSSION TITLE
Logout confirmation shown even with included OIDC keycloak.js adapter

### DISCUSSION THREAD

**Discussion by lazka** [keycloak/keycloak#12114] [Q&A] [answered]
After updating to Keycloak 18 there is now a "Do you want to logout?" page when logging out via the official Javascript adapter as provided by the Keycloak instance. Logging out +  redirect works fine, but the user has to press an extra button.

The changelog says:

> "It is possible to omit the confirmation and do automatic redirect to the application when you
> include parameter post_logout_redirect_uri together with the parameter id_token_hint with the ID
> Token used for login."

From what I see the Javascript adapter properly sets post_logout_redirect_uri as well as id_token_hint here, but still the "Do you want to logout?" page is shown.

Is there anything else I need to do to get rid of the confirmation page?

thanks

**Comment by extemporis** [comment, marked as answer]
It seems that the activated client option "Consent Required"  is causing this behavior. Without that the logout is working as expected.

**Reply by lazka** [reply]
Thanks, that indeed works. But also disables the login consent page :(

**Reply by extemporis** [reply]
Yes, this seems to be a bug: https://github.com/keycloak/keycloak/issues/11915

**Reply by lazka** [reply]
I'll subscribe there then and mark this as solved

thanks 🙏

**Reply by romantaranenko-hometogo** [reply]
Can you please specify where you found that option, since the only "Consent required" I found is under client settings and in my case it's disabled

**Reply by marcellobarile** [reply]
same here, "Constent required" is disabled and the logout contains the redirectUrl, but the confirmation is still present.

**Comment by ddubrava** [comment]
The logout confirmation is shown if you do not pass the "id_token_hint", which is created from the "idToken".

https://github.com/keycloak/keycloak/blob/bcab75a7ef19a5a7edc0944aa36ad910c29204b9/js/libs/keycloak-js/src/keycloak.js#L486-L488

To clarify, the "id_token_hint" is generated from the "idToken" and is required to confirm the logout process. If you manage tokens on your side, such as storing them in a local storage, and pass them to the Keycloak init method, you need to provide not only the token and refreshToken but also the idToken. This enables Keycloak to create the necessary "id_token_hint" during logout.

However, a problem arises when you save tokens in local storage because all tabs share the same tokens. So, if you log out in one tab and then attempt to log out in another tab, you will encounter the logout confirmation screen. This happens because Keycloak cannot create the "id_token_hint" in this scenario.

On the other hand, if you do not manage tokens on your side (i.e., you do not save or pass them to keycloak.init), you should not experience these problems.

By the way, there might be a bug with the logout confirmation screen, even if you do not manage tokens on your side. For instance, if you log in on both tabs, log out in the first tab, and then log in again, and then log out in the second tab, you will see the confirmation screen.

You can refer to the following resources for more information on this topic:

1. https://github.com/keycloak/keycloak/issues/12002#issuecomment-1128896662
2. https://github.com/keycloak/keycloak/discussions/12183
3. https://stackoverflow.com/questions/75686708/keycloak-redirect-to-confirm-logout-page-when-logging-out-after-a-page-refre

**Reply by Walnussbaer** [reply]
> By the way, there might be a bug with the logout confirmation screen, even if you do not manage tokens on your side. For instance, if you log in on both tabs, log out in the first tab, and then log in again, and then log out in the second tab, you will see the confirmation screen.

@ddubrava 
So is there a bug ticket for this behaviour? We just encountered this behaviour on Keycloak 26.2.5. It's not a big deal breaker, I am just curious whether this is something that will be fixed eventually. We do not manage tokens on our side manually, so no identity token is passed during the adapter initialization.

**Reply by ddubrava** [reply]
Hey! I’m not a maintainer, just a user. So, cannot share any insights on this. It’s better to ping maintainers

**Reply by Walnussbaer** [reply]
> Hey! I’m not a maintainer, just a user. So, cannot share any insights on this. It’s better to ping maintainers

Funny, your post looked like it was written by some maintainer. Didn't event question it haha. But thanks for the reply
